src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```diff
@@ -109,6 +109,11 @@ public KestrelHttpParser(IKestrelTrace log)
 
                     pathEncoded = true;
                 }
+                else if (ch == 0)
```

Comment:
We are validating these during materialization of values, fail fast is less important then less checks in hot paths